connector/src/main/java/org/geysermc/connector/utils/CooldownUtils.java

```diff
@@ -0,0 +1,119 @@
+
+    static {
+        SHOW_COOLDOWN = GeyserConnector.getInstance().getConfig().isShowCooldown();
+    }
+
+    /**
+     * Starts sending the fake cooldown to the Bedrock client.
+     * @param session GeyserSession
+     */
+    public static void sendCooldown(GeyserSession session) {
+        if (!SHOW_COOLDOWN) return;
+        if (session.getAttackSpeed() == 0.0 || session.getAttackSpeed() > 20) return; // 0.0 usually happens on login and causes issues with visuals; anything above 20 means a plugin OldCombatMechanics is being used
+        // Needs to be sent or no subtitle packet is recognized by the client
+        SetTitlePacket titlePacket = new SetTitlePacket();
+        titlePacket.setType(SetTitlePacket.Type.SET_TITLE);
+        titlePacket.setText(" ");
+        session.sendUpstreamPacket(titlePacket);
+        long lastHitTime = System.currentTimeMillis();
+        computeCooldown(session, lastHitTime);
+    }
+
+    /**
+     * Keeps updating the cooldown until the bar is complete.
+     * @param session GeyserSession
+     * @param lastHitTime The time of the last hit. Used to gauge how long the cooldown is taking.
+     */
+    private static void computeCooldown(GeyserSession session, long lastHitTime) {
+        if (session.isClosed()) return; // Don't run scheduled tasks if the client left
+        SetTitlePacket titlePacket = new SetTitlePacket();
+        titlePacket.setType(SetTitlePacket.Type.SET_SUBTITLE);
+        titlePacket.setText(getTitle(session, lastHitTime));
+        titlePacket.setFadeInTime(0);
+        titlePacket.setFadeOutTime(5);
+        titlePacket.setStayTime(2);
+        session.sendUpstreamPacket(titlePacket);
+        if (hasCooldown(session, lastHitTime)) {
+            session.getConnector().getGeneralThreadPool().schedule(() -> computeCooldown(session, lastHitTime), 50, TimeUnit.MILLISECONDS); // Updated per tick. 1000 divided by 20 ticks equals 50
+        } else {
+            SetTitlePacket removeTitlePacket = new SetTitlePacket();
+            removeTitlePacket.setType(SetTitlePacket.Type.SET_SUBTITLE);
+            removeTitlePacket.setText(" ");
+            session.sendUpstreamPacket(removeTitlePacket);
+        }
+    }
+
+    private static boolean hasCooldown(GeyserSession session, long lastHitTime) {
+        long time = System.currentTimeMillis() - lastHitTime;
+        double cooldown = restrain(((double) time) * session.getAttackSpeed() / 1000d, 1.5);
+        return cooldown < 1.1;
+    }
+
+
+    private static double restrain(double x, double max) {
+        if (x < 0d)
+            return 0d;
+        return Math.min(x, max);
+    }
+
+    private static String getTitle(GeyserSession session, long lastHitTime) {
+        long time = System.currentTimeMillis() - lastHitTime;
+        double cooldown = restrain(((double) time) * session.getAttackSpeed() / 1000d, 1);
+
+        int darkGrey = (int) Math.floor(((double) 10) * cooldown);
```

Comment:
Is there a difference in having `((double) 10)` instead of just `10d`? Seems like a pointless cast